frontend/src/lib/hooks.ts

```diff
@@ -0,0 +1,42 @@
+import { ref } from 'vue';
+import type { Ref } from 'vue';
+import type { AxiosRequestConfig } from 'axios';
+
+import { makeRequest } from '@/lib/requests';
+import type { ApiResponse } from '@/lib/requests';
+
+export interface UseFetchReturn<T> {
+    data: Ref<T | null>;
+    error: Ref;
+    isFetching: Ref<boolean>;
+}
+
+type UseFetchOptions = Omit<AxiosRequestConfig, 'url'>;
+
+export async function useFetch<TData extends ApiResponse>(
+    url: string,
+    options?: UseFetchOptions
+): Promise<UseFetchReturn<TData>> {
+    const state: UseFetchReturn<TData> = {
```

Comment:
Another alternative might be using `reactive`. See https://medium.com/swlh/vue3-using-ref-or-reactive-88d47c8f6944

Reply:
I saw this option too, but I found less clear

Reply:
It's cleaner ^^ I think. But I'm fine with both

Reply:
Let's keep `ref`s, then 😇